src/aind_data_schema/utils/validators.py

```diff
@@ -106,3 +108,26 @@ def recursive_get_all_names(obj: Any) -> List[str]:
             names.extend(recursive_get_all_names(field_value))
 
     return names
+
+
+def recursive_check_paths(obj: Any, directory: Optional[Path] = None):
+    """
+    Recursively checks if any attribute in an object is a pathlib.Path and logs if it doesn't exist.
+    """
+    if isinstance(obj, Enum):
+        return
+
+    if isinstance(obj, PurePosixPath):
+        obj = Path(obj)
+
+    if isinstance(obj, Path):
```

Comment:
This is what I thought we want:
```suggestion
    if isinstance(obj, AssetPath):
```

Reply:
that'll work fine for `directory / obj` but not for `obj.exists`, so we'll need to remove the directory=None option, which doesn't seem to be used anyway.

Reply:
AssetPath as defined has no exists() function, we have to convert to Path to use the local filesystem.

Reply:
true, what i meant was `directory / obj` should be a Path. explicitly converting is fine too, but it should all be nested under a top `isinstance(obj, AssetPath)`

Reply:
If we do type checks against AssetPath we have to move AssetPath into its own file (recursive imports), I'm inclined to keep it in identifiers. I think it's fine as-is, unless I'm missing something here you're not changing functionality just how the code is written?

Reply:
It is different functionally if a user inserts a PurePosixPath somewhere random, but I'm not sure if that actually matters. can't you just add the import inline within the function here though?

Reply:
Okay now I understand. You couldn't add the import it was circular, but I refactored it so that this works.

Reply:
I see now. A schema developer would need to add a PurePosixPath typed field to trigger that issue, but there's no reason to risk it I refactored it so that the import works.